### ISSUE TITLE
Bug in ps_facetedsearch v. 3.3.0 - Value of an attribute = Value of a feature

### ISSUE THREAD

**Issue by Didweb** [PrestaShop/PrestaShop#15635]
Bug in the **ps_facetedsearch** module. **v3.3.0**

When characteristics and attributes with the same name are created.

The Features do not filter the results well.

**To Reproduce**

-  Create an attribute "color" with value "grey"
-  Create a feature "color" with value "grey"
-  Create a product with combination with attribute grey
-  Create another product with feature grey
-  Go in FO and select feature "grey" => No results
-  Go in FO and select attribute "grey" => No results

**Comment by omar2886** [comment]
Filter by price range shows no products after upgrading to this version, the other wilters work

**Comment by Didweb** [comment]
I have tried to eliminate the filter by prices and have Attributes and Characteristics with the same name produces the error (with the characteristics): It does not show products.

On the other hand, if I change the name of the characteristics if I perform the search.

**Comment by PierreRambaud** [comment]
@omar2886 You need to rebuild price indexes.

**Comment by khouloudbelguith** [comment]
Hi @Didweb,

I manage to reproduce the issue with PS1.7.6.1 & ps_facetedsearch v2.2.1 / v3.0.6 / v3.3.0.
Also the same issue with PS1.7.5.2 & ps_facetedsearch v3.3.0.
https://drive.google.com/file/d/1F2LVf_aBbloEKnzQ1gcn6muaYz9gevC3/view
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by omar2886** [comment]
Hi @PierreRambaud & @khouloudbelguith 
when rebuilding index it gives me an error, also if I first clear cache (indexation failed). I used to have this same error on a previous version of this module, I think a specific caching folder must be manually erased, but don't remember, can you remind me please?

**Comment by PierreRambaud** [comment]
The cache is saved in database. Have you try to reinstall the module?

**Comment by omar2886** [comment]
@PierreRambaud just did it, had to reconfigure it all, still get the same error during indexation. But! I found out that the filter by price (asc. desc.) works when you have applied first another filter from the facet menu, i.e. if you first filter a brand/condition/availability... then apply the filter by price, it works. Bizarre!
Still buggie and non functional, please release a new upgrade ASAP and this time include the #14518

**Comment by marionf** [comment]
@omar2886 Your issue seems different of this one, please open a new issue.
@Didweb Can you explain why your characteristics and attributes have the same name ?
I don't see the interest of doing this

**Comment by Didweb** [comment]
@marionf 
Initially, my client was not very clear on how to assign characteristics and variety of products.

And provisionally they assume duplicate options. Then notice the strange behavior of the pro faceted seeker. Until I realized that it was because the names were the same.

This may be the case, for example:
Some products have color combinations and other products do not have combinations, they are simply assigned a color as the product's characteristic.

Then the two options Color (combinations) and color (characteristic) come out. That in my opinion is also an error and that duplicates the search. But the bad thing is the characteristic does not filter well.

**Comment by PierreRambaud** [comment]
@marionf Could you tell us what do you expect here?
Because Faceted search is building URL like this:
http://prestashop/gb/6-accessories?q=Color-Black

Should we change it?

**Comment by marionf** [comment]
Is it possible to change the parameters in the URL ?
For example like this, where "color" & "size" are the feature and attribute's names:
feature_color=grey
attribute_size=S
feature_size=S
attribute_color=grey

**Comment by PierreRambaud** [comment]
Yes, we can change the URL, but it will result in SEO errors because old URLs are already fetched by Google and other search engines.

So the URL will look like:
http://prestashop/gb/6-accessories?q=Feature_Color-Black/Attribute_Color-Black
http://prestashop/gb/6-accessories?q=Caractéristique_Color-Black/Attribut_Color-Black

It looks a little bit weird but it can be a solution :thinking:

**Comment by PierreRambaud** [comment]
Hum I will have to change how URLs are build because of this bug #16320 :sweat_smile:

**Comment by PierreRambaud** [comment]
Could we imagine something like this instead: `?q[Category][]=Men&q[Feature][Color][]=White&q[Attribute][Color][]=Black` ?

**Comment by Hlavtox** [comment]
@PierreRambaud Hi, I was just debugging why there is an empty page when I select a color from filters. Ran the query manually and wondered why there is an attribute condition in the query. Aaaaahaaaaa. Renamed Color feature to something else and now it works. We need to fix this man!!! :D

I tried just to change the feature URL slug, but also didn't work. Is it even used or is it a remain of layered search?

**Comment by boo-code** [comment]
Still reproduces on ps_facetedsearch 5.0.0 / PrestaShop 9.2.0. Fixture: attribute group "Color" with value "White" (4 products), plus a new feature "Color" with value "White" (1 product), both filters enabled on the same category, index rebuilt.

```
?q=Color-White          No products     <- the bug
?q=Color-Black          4 products      <- control, attribute value with no colliding feature value
?q=Composition-Cotton   2 products      <- control, a feature filter on its own works
```

@Hlavtox's note that changing the feature's URL slug "also didn't work" is reproducible, and there is a specific reason for it. The two halves of the round trip are not symmetrical:

- `URLSerializer::getFacetLabel()` and `getFilterLabel()` already prefer `url_name`, and the generated links do switch once a slug is set — measured, the facet link became `q=feature%5C-color-White` and following it returns the feature's 1 product correctly.
- `Converter::createFacetedSearchFiltersFromQuery()` reads

```php
if (isset($receivedFilters[$feature['url_name']])) { ... }
elseif (isset($receivedFilters[$feature['name']])) { ... }
```

so the feature keeps matching the attribute group's URL segment whatever slug it has. `?q=Color-White` therefore activates both filters, they AND, and the page is empty. The attribute group branch a few lines below has the same fallback.

Dropping the plain-name fallback for a facet that has an explicit slug closes it: `?q=Color-White` returns the 4 attribute matches and `?q=<slug>-White` returns the feature's 1. Measured with a control that matters here — "Composition" has no slug configured and still resolves by its plain name, so URLs already indexed for shops that never set one keep working. `DataAccessor::getFeatures()` and `getAttributesGroups()` already normalise an empty slug to `NULL`, so the distinction is available without a schema change.

That is narrower than PrestaShop/ps_facetedsearch#779, which changed the URL scheme to numeric IDs, declared a BC break and was closed unmerged in November 2023. The cost here is limited to shops that have deliberately configured a slug, whose plain-name URLs are already orphaned by the link generation.
